File: src/main/java/dev/esophose/discordbot/command/commands/info/CommandInfoCommand.java
package dev.esophose.discordbot.command.commands.info;

import dev.esophose.discordbot.Sparky;
import dev.esophose.discordbot.command.DiscordCommand;
import dev.esophose.discordbot.command.DiscordCommandMessage;
import dev.esophose.discordbot.manager.CommandManager;
import dev.esophose.discordbot.manager.GuildSettingsManager;
import discord4j.core.object.util.Permission;
import discord4j.core.object.util.PermissionSet;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class CommandInfoCommand extends DiscordCommand {

    public CommandInfoCommand(Sparky bot) {
        super(bot);
    }

    public void execute(DiscordCommandMessage message, DiscordCommand command) {
        CommandManager commandManager = this.bot.getManager(CommandManager.class);
        String commandPrefix = this.bot.getManager(GuildSettingsManager.class).getGuildSettings(message.getGuildId()).getCommandPrefix();

        String info =
                "**Command Name:** " + command.getName() + '\n' +
                        "**Aliases:** " + "[" + String.join(", ", command.getAliases()) + "]" + '\n' +
                        "**Description:** " + command.getDescription() + '\n' +
                        "**Parameters:** " + commandManager.getCommandUsage(command, false, commandPrefix) + '\n' +
                        "**Required Member Permission:** " + command.getRequiredMemberPermission(message.getGuildId()).name() + '\n' +
                        "**Required Bot Permissions:** " + "[" + command.getRequiredBotPermissions().stream().map(Enum::name).collect(Collectors.joining(", ")) + "]";
        commandManager.sendResponse(message.getChannel(), "Info for ." + command.getName(), info).subscribe();
    }

    @Override
    public String getName() {
        return "cinfo";
    }

    @Override
    public List<String> getAliases() {
        return Collections.singletonList("commandinfo");
    }

    @Override
    public String getDescription() {
        return "Displays info for a command";
    }

    @Override
    public PermissionSet getRequiredBotPermissions() {
        return PermissionSet.of(Permission.SEND_MESSAGES);
    }

    @Override
    public Permission getDefaultRequiredMemberPermission() {
        return Permission.SEND_MESSAGES;
    }

}

File: src/main/kotlin/dev/esophose/discordbot/command/commands/info/CommandInfoCommand.kt
package dev.esophose.discordbot.command.commands.info

import dev.esophose.discordbot.Sparky
import dev.esophose.discordbot.command.DiscordCommand
import dev.esophose.discordbot.command.DiscordCommandMessage
import dev.esophose.discordbot.manager.CommandManager
import dev.esophose.discordbot.manager.GuildSettingsManager
import discord4j.core.`object`.util.Permission
import discord4j.core.`object`.util.PermissionSet
import java.util.stream.Collectors

class CommandInfoCommand : DiscordCommand() {

    override val name: String
        get() = "cinfo"

    override val aliases: List<String>
        get() = listOf("commandinfo")

    override val description: String
        get() = "Displays info for a command"

    override val requiredBotPermissions: PermissionSet
        get() = PermissionSet.of(Permission.SEND_MESSAGES)

    override val defaultRequiredMemberPermission: Permission
        get() = Permission.SEND_MESSAGES

    fun execute(message: DiscordCommandMessage, command: DiscordCommand) {
        val commandManager = Sparky.getManager(CommandManager::class)
        val commandPrefix = Sparky.getManager(GuildSettingsManager::class).getGuildSettings(message.guildId).commandPrefix

        val info = "**Command Name:** " + command.name + '\n'.toString() +
                "**Aliases:** " + "[" + command.aliases.joinToString(", ") + "]" + '\n'.toString() +
                "**Description:** " + command.description + '\n'.toString() +
                "**Parameters:** " + commandManager.getCommandUsage(command, false, commandPrefix) + '\n'.toString() +
                "**Required Member Permission:** " + command.getRequiredMemberPermission(message.guildId).name + '\n'.toString() +
                "**Required Bot Permissions:** " + "[" + command.requiredBotPermissions.stream().map { it.name }.collect(Collectors.joining(", ")) + "]"
        commandManager.sendResponse(message.channel, "Info for ." + command.name, info).subscribe()
    }

}
